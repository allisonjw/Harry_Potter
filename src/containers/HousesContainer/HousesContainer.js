import React from 'react';
import './HousesContainer.scss';
// import CharactersContainer from '../CharacterContainer/CharacterContainer';
import HouseCard from '../HouseCard/HouseCard';
import { connect } from 'react-redux';


export const HousesContainer = props => {
   const { allHouses } = props
 
    const displayHouses = allHouses.map(house => {
            return (
          <HouseCard
          {...house}
            key={house.id}
          />
          )
        })
    

        return (
              <section className="section_house-main">
                {displayHouses}
              </section>
        )
    }


export const mapStateToProps = state => ({
    allHouses: state.allHouses, 
});

export default connect(mapStateToProps)(HousesContainer);
