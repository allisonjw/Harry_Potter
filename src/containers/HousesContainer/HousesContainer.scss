.section_house-main {
    background-image: url('https://happytowander.com/wp-content/uploads/Gloucester-Cathedral-in-England-1280-2.jpg');
    background-size: cover;
}

.section_houses {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    overflow-x: scroll;
    overflow-y: hidden;
    padding: 1%;
}

.section_houses-heading {
    display: flex;
    flex-direction: row;
    font-size: 2em;
    height: 20%;
    justify-content: space-evenly;
    margin-bottom: 3%;
    width: 100%;
      .houses_heading {
        color: white;
        text-shadow: 2px 2px 2px black;
        margin-right: 15%;
        margin-top: 3%;
      }
}

.back_btn {
    border-radius: 3px;
    font-size: 1.2em;
    height: 60px;
    width: 100%;
    margin: 20% 0 0 15%;
    &:hover {
        cursor: pointer;
        transform: scale(1.1);
        transition: all .4s ease-in-out;
      }
}