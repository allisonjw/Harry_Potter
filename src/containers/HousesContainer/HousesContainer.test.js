import React from 'react';
import { shallow } from 'enzyme';
import { HousesContainer } from '../HousesContainer/HousesContainer';

describe('HousesContainer', () => {
    const mockAllHouses = [
        {
            _id: "5a05e2b252f721a3cf2ea33f",
            name: "Gryffindor",
            mascot: "lion",
            headOfHouse: "Minerva McGonagall",
            houseGhost: "Nearly Headless Nick",
            founder: "Goderic Gryffindor",
            __v: 0,
            school: "Hogwarts School of Witchcraft and Wizardry",
            members: [
            "5a0fa648ae5bc100213c2332",
            "5a0fa67dae5bc100213c2333"],
            values: [
                "courage",
                "bravery",
                "nerve",
                "chivalry"
                ],
                colors: [
                "scarlet",
                "gold"
                ]
        },
        {
            _id: "5a05da69d45bd0a11bd5e06f",
            name: "Ravenclaw",
            mascot: "eagle",
            headOfHouse: "Fillius Flitwick",
            houseGhost: "The Grey Lady",
            founder: "Rowena Ravenclaw",
            __v: 0,
            school: "Hogwarts School of Witchcraft and Wizardry",
            members: [
            "5a0fa8a6ae5bc100213c233b",
            "5a107ffee0686c0021283b21"],
            values: [
                "intelligence",
                "creativity",
                "learning",
                "wit"
                ],
                colors: [
                "blue",
                " bronze"
                ]
        }
      ];

      const mockAddHouses = [
        { 
            _id: "5a05e2b252f721a3cf2ea33f",
            name: "Gryffindor",
            mascot: "lion",
            headOfHouse: "Minerva McGonagall",
            houseGhost: "Nearly Headless Nick",
            founder: "Goderic Gryffindor",
            __v: 0,
            school: "Hogwarts School of Witchcraft and Wizardry",
            members: [
            "5a0fa648ae5bc100213c2332",
            "5a0fa67dae5bc100213c2333"],
            values: [
                "courage",
                "bravery",
                "nerve",
                "chivalry"
                ],
                colors: [
                "scarlet",
                "gold"
                ]
        }
      ];

    const wrapper = shallow(<HousesContainer  
        allHouses={mockAllHouses}
        addHouses={mockAddHouses}/>

);

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
});