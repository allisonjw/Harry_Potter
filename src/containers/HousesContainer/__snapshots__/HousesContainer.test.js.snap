// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HousesContainer should match snapshot with all information passing in correctly 1`] = `
<section
  className="section_house-main"
>
  <section
    className="section_houses-heading"
  >
    <Link
      to="/login"
    >
      <button
        className="back_btn"
      >
        Back to Login
      </button>
    </Link>
    <h2
      className="houses_heading"
    >
      Hogwarts Houses
    </h2>
  </section>
  <section
    className="section_houses"
  >
    <Connect(HouseCard)
      __v={0}
      _id="5a05e2b252f721a3cf2ea33f"
      colors={
        Array [
          "scarlet",
          "gold",
        ]
      }
      founder="Goderic Gryffindor"
      headOfHouse="Minerva McGonagall"
      houseGhost="Nearly Headless Nick"
      key="5a05e2b252f721a3cf2ea33f"
      mascot="lion"
      members={
        Array [
          "5a0fa648ae5bc100213c2332",
          "5a0fa67dae5bc100213c2333",
        ]
      }
      name="Gryffindor"
      school="Hogwarts School of Witchcraft and Wizardry"
      values={
        Array [
          "courage",
          "bravery",
          "nerve",
          "chivalry",
        ]
      }
    />
    <Connect(HouseCard)
      __v={0}
      _id="5a05da69d45bd0a11bd5e06f"
      colors={
        Array [
          "blue",
          " bronze",
        ]
      }
      founder="Rowena Ravenclaw"
      headOfHouse="Fillius Flitwick"
      houseGhost="The Grey Lady"
      key="5a05da69d45bd0a11bd5e06f"
      mascot="eagle"
      members={
        Array [
          "5a0fa8a6ae5bc100213c233b",
          "5a107ffee0686c0021283b21",
        ]
      }
      name="Ravenclaw"
      school="Hogwarts School of Witchcraft and Wizardry"
      values={
        Array [
          "intelligence",
          "creativity",
          "learning",
          "wit",
        ]
      }
    />
  </section>
</section>
`;
